<a href="https://colab.research.google.com/github/GenoKiller777/PythonBasico/blob/main/ElnegocioAccesoInternet_DarwinJoanPerezAvendano.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Considere el siguiente enunciado:
* En la provincia de Soria, el negocio de acceso a Internet se reparte entre dos operadores, **Timofónica** y **Robafone** y dos únicas marcas de routers, **Xisco** y **Nuaweii**.
* En Soria, la cuota de mercado de **Timofónica** es del **60%** y de **Robafone** el resto (**40%**).
* El **70%** de los usuarios dispone de router **Xisco** y el **30%**
de **ambas marcas**.
* Además se sabe que la _probabilidad_ de corte de acceso es **0.1** para usuarios de **Timofónica**, **0.15** para **Robafone** y **0.05** para routers **Xisco**.

### **1.** Crear un script en Python que defina las probabilidades proporcionadas en el enunciado (usar un diccionario).


In [1]:
pr = dict()
pr["Timofónica"] = 0.6
pr["Robafone"] = 0.4
pr["Xisco"] = 0.7
pr["Xisco Y Nuaweii"] = 0.3
pr["Corte | Timofónica"] = 0.1
pr["Corte | Robafone"] = 0.15
pr["Corte | Xisco"] = 0.05

### **2.** Implementar en el script las expresiones correspondiente al Teorema de Bayes para responder las siguientes preguntas:


#### • ¿Cuál es la probabilidad de que a un usuario se le corte el Internet?

In [21]:
pr["Corte"] =  pr["Corte | Timofónica"] * pr["Timofónica"] + pr["Corte | Robafone"] * pr["Robafone"]
print(round(pr["Corte"] * 100 , 3),"%")

12.0 %


**R.-** _La probabilidad de que un usuario  tenga un corte es del **12%**._


#### • Si se sabe que un usuario tiene la línea cortada, ¿cuál es la probabilidad de que tenga router Xisco en casa?,


In [22]:
pr["Xisco | Corte"] = pr["Xisco"] * pr["Corte | Xisco"] / pr["Corte"]
print(round(pr["Xisco | Corte"] * 100, 3),"%")

29.167 %


**R.-** _Si el usuario tiene un corte, la probabilidad de que en su casa tenga un router **Xisco** es del **29.167%**._

#### • ¿Cuál es la probabilidad de que se produzca un corte a un usuario que no tiene un router Xisco?

In [23]:
pr["Corte | Nuaweii"] = (pr["Corte"] - pr["Xisco"] * pr["Corte | Xisco"]) / (1 - pr["Xisco"])
print(round(pr["Corte | Nuaweii"] * 100, 3),"%")

28.333 %


**R.-** _La probabilidad de que un usuario tenga un corte y no tenga un router **Xisco** es del **28.333%**._